mavros/src/plugins/global_position.cpp

```diff
@@ -227,16 +227,20 @@ class GlobalPositionPlugin : public plugin::PluginBase {
 		g_origin->header.frame_id = tf_global_frame_id;
 		g_origin->header.stamp = ros::Time::now();
 
+		g_origin->position.latitude = glob_orig.latitude / 1E7;
```

Comment:
No need. Just move `glob_orig.altitude / 1E3 + m_uas->geoid_to_ellipsoid_height(&g_origin->position)` to https://github.com/mavlink/mavros/pull/1298/files#diff-cefbff5235993719761a0d6a14e67129R242

Reply:
Nevermind. You need to convert Lat and Long first. I guess this works.